src/core/net/tcp6.hpp

```diff
@@ -329,7 +332,32 @@ class Tcp : public InstanceLocator, private NonCopyable
          */
         Error Deinitialize(void);
 
+        /**
+         * Converts a reference to a struct tcpcb to a reference to its
+         * enclosing Endpoint.
+         */
+        static Endpoint &FromTcb(struct tcpcb &aTcb) { return *reinterpret_cast<Endpoint *>(&aTcb); }
+
+        /**
+         * Obtains a reference to this Endpoint's struct tcpcb.
+         */
+        struct tcpcb &GetTcb(void) { return *reinterpret_cast<struct tcpcb *>(&mTcb); }
+
+        /**
+         * Obtains a const reference to this Endpoint's struct tcpcb.
+         */
+        const struct tcpcb &GetTcb(void) const { return *reinterpret_cast<const struct tcpcb *>(&mTcb); }
+
+        /**
+         * Checks if this Endpoint is in the closed state.
+         */
+        bool IsClosed(void) { return GetTcb().t_state == TCP6S_CLOSED; }
```

Comment:
```suggestion
        bool IsClosed(void) const { return GetTcb().t_state == TCP6S_CLOSED; }
```

Reply:
Thanks. This is fixed in the latest update.